front/lib/api/assistant/configuration.ts

```diff
@@ -204,47 +226,160 @@ export async function getAgentConfiguration(
+    }
+    const agentIds = (
+      await AgentConfiguration.findAll({
+        where,
+        attributes: ["sId"],
+      })
+    ).map((a) => a.sId);
+    return [...(await getLocalAgentsFromIds(agentIds)), ...globalAgents];
+  }
+
+  if (agentsGetView === "all") {
+    const agentIds = (
+      await AgentConfiguration.findAll({
+        where: {
+          ...where,
+          scope: { [Op.in]: ["published", "workspace"] },
+        },
+        attributes: ["sId"],
+      })
+    ).map((a) => a.sId);
+    return [...(await getLocalAgentsFromIds(agentIds)), ...globalAgents];
+  }
+
+  if (agentsGetView === "list") {
+    const user = auth.user();
+    if (!user) {
+      throw new Error("List view is specific to a user.");
+    }
+    const agentIds = (
+      await AgentConfiguration.findAll({
+        where: {
+          ...where,
+          [Op.or]: [
+            { scope: { [Op.in]: ["published", "workspace"] } },
+            { authorId: user.id },
+          ],
+        },
+        attributes: ["sId", "scope", "name"],
+      })
+    ).map((a) => a.sId);
+    const localAgents = (await getLocalAgentsFromIds(agentIds)).filter((a) => {
+      if (a.scope === "workspace") {
+        return a.relationOverride !== "not-in-list";
+      }
+      if (a.scope === "published") {
+        return a.relationOverride === "in-list";
+      }
+      return true; // user's private agents should be returned
+    }) as AgentConfigurationType[];
+    return [...localAgents, ...globalAgents];
+  }
+
+  // conversation view
+  if (typeof agentsGetView === "object" && agentsGetView.conversationId) {
+    const user = auth.user();
+    if (!user) {
+      throw new Error("Conversation view is specific to a user.");
+    }
+    const [agents, mentions] = await Promise.all([
+      AgentConfiguration.findAll({
+        where: {
+          ...where,
+          [Op.or]: [
+            { scope: { [Op.in]: ["published", "workspace"] } },
+            { authorId: user && user.id },
+          ],
+        },
+        attributes: ["sId", "scope", "name"],
+      }),
+      getMentions(agentsGetView.conversationId),
+    ]);
+    const agentIds = agents.map((a) => a.sId);
+    const mentionedAgentIds = mentions.map((m) => m.configurationId);
+    const localAgents = (await getLocalAgentsFromIds(agentIds)).filter((a) => {
+      if (mentionedAgentIds.includes(a.sId)) {
+        return true;
+      }
+      if (a.scope === "workspace") {
+        return a.relationOverride !== "not-in-list";
+      }
+      if (a.scope === "published") {
+        return a.relationOverride === "in-list";
+      }
+      return true; // user's private agents should be returned
+    }) as AgentConfigurationType[];
+    return [...localAgents, ...globalAgents];
+  }
+
+  throw new Error(`Unknown agentsGetView ${agentsGetView}`);
+}
+
+async function getMentions(conversationId: string): Promise<AgentMention[]> {
+  const mentions = await Mention.findAll({
```

Comment:
Same here this query is complex. We want to make sure it's cost is low :+1:

Reply:
Do we we want `messageId, agentConfigurationId` as an index?

Reply:
Not necessary IMHO, the query planner uses the where appropriately
```
SELECT "mention"."id", "mention"."agentConfigurationId" FROM "mentions" AS "mention" LEFT OUTER JOIN ( "messages" AS "message" INNER JOIN "conversations" AS "message->conversation" ON "message"."conversationId" = "message->conversation"."id" AND "message->conversation"."sId" = '9c81201e61' ) ON "mention"."messageId" = "message"."id" WHERE "mention"."agentConfigurationId" IS NOT NULL;

---------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Hash Left Join  (cost=16.34..28.27 rows=139 width=520) (actual time=0.081..0.090 rows=7 loops=1)
   Hash Cond: (mention."messageId" = message.id)
   ->  Seq Scan on mentions mention  (cost=0.00..11.40 rows=139 width=524) (actual time=0.011..0.013 rows=7 loops=1)
         Filter: ("agentConfigurationId" IS NOT NULL)
   ->  Hash  (cost=16.33..16.33 rows=1 width=4) (actual time=0.061..0.062 rows=2 loops=1)
         Buckets: 1024  Batches: 1  Memory Usage: 9kB
         ->  Nested Loop  (cost=0.29..16.33 rows=1 width=4) (actual time=0.037..0.039 rows=2 loops=1)
               ->  Index Scan using conversations_s_id on conversations "message->conversation"  (cost=0.14..8.16 rows=1 width=4) (actual time=0.021..0.022 rows=1 loops=1)
                     Index Cond: (("sId")::text = '9c81201e61'::text)
               ->  Index Scan using messages_conversation_id_rank_version on messages message  (cost=0.14..8.16 rows=1 width=8) (actual time=0.012..0.013 rows=2 loops=1)
                     Index Cond: ("conversationId" = "message->conversation".id)
 Planning Time: 0.711 ms
 Execution Time: 0.135 ms
```

Reply:
:+1: